js/dialog.js

```diff
@@ -37,6 +37,23 @@ export async function setupDetailsDialog() {
   });
 }
 
+function attackTraitsHTML(cost, damage, type) {
+  if (!cost && !damage && !type) {
+    return "";
+  }
+  const htmlList = [];
+  if (type) {
+    htmlList.push(`Type: ${type}`);
+  }
+  if (cost) {
+    htmlList.push(`Cost: ${cost}`);
+  }
+  if (damage) {
+    htmlList.push(`Damage: ${damage}`);
+  }
+  return `<p>${htmlList.filter((item) => Boolean(item)).join(" | ")}</p>`;
+}
+
```

Comment:
There's a good way to compress this: Instead of receiving individual parameters, you can receive an object with the name of the variables as keys. That way you can use object traversal methods which can eliminate the repetition.

Reply:
hmm can you customize the label ('Type:') easily though? I'll go ahead and merge and leave this in your capable hands!